GDJS/Runtime/variable.ts

```diff
@@ -136,73 +207,76 @@ namespace gdjs {
      * @param childName The name of the child to be removed
      */
     removeChild(childName: string) {
-      if (!this._isStructure) {
-        return;
-      }
+      if (this._type !== 'structure') return;
       delete this._children[childName];
     }
 
     /**
      * Remove all the children.
-     *
-     * If the variable is not a structure, nothing is done.
      */
     clearChildren() {
-      if (!this._isStructure) {
-        return;
-      }
-      for (const child in this._children) {
-        if (this._children.hasOwnProperty(child)) {
-          delete this._children[child];
-        }
-      }
+      this._children = {};
+      this._childrenList = [];
     }
 
     /**
      * Replaces all the children with a new map of children.
      * @param newChildren The map of new children.
      */
-    replaceChildren(newChildren: { [key: string]: gdjs.Variable }) {
-      this._isStructure = true;
+    replaceChildren(newChildren: Children) {
+      this._type = 'structure';
       this._children = newChildren;
     }
 
+    /**
+     * Replaces all the children with a new array of children.
+     * @param {gdjs.Variable[]} newChildren The array of new children.
+     */
+    replaceChildrenList(newChildren: gdjs.Variable[]) {
```

Comment:
Same as in the C++ equivalent, I think it might be a good idea to call this `replaceChildrenArray`? Might be a bit more explicit for the reader/user: "ok so they have structure, array... but what is this list thing? Oh it's array again!"